repository/src/main/java/org/apache/atlas/discovery/EntityDiscoveryService.java

```diff
@@ -1046,6 +1049,28 @@ public AtlasSearchResult directIndexSearch(SearchParams searchParams) throws Atl
         return ret;
     }
 
+    private String getIndexName(IndexSearchParams params) throws AtlasBaseException {
+        if (StringUtils.isEmpty(params.getPersona()) && StringUtils.isEmpty(params.getPurpose())) {
+            return INDEX_PREFIX + VERTEX_INDEX;
+        }
+
+        String qualifiedName = "";
+        if (StringUtils.isNotEmpty(params.getPersona())) {
+            qualifiedName = params.getPersona();
+        } else {
+            qualifiedName = params.getPurpose();
+        }
+
+        String[] parts = qualifiedName.split("/");
+        String aliasName = parts[parts.length - 1];
```

Comment:
Can `trim` be used instead of this?

Reply:
no, qualifiedName will have format like <tenant_id/UUID_string>, where UUID_string is the ES alias name